src/types.ts

```diff
@@ -101,3 +101,45 @@ export interface Column {
   stylingInfo: string[];
   sortDirection: string;
 }
+
+export type TableData =
+  | {
+      totalColumnCount: number;
+      totalRowCount: number;
+      totalPages: number;
+      paginationNeeded: boolean;
+      rows: number[];
+      columns: number[];
```

Comment:
Pretty sure they are not arrays with numbers. There is already a type for columns, might have to define the rows one